modyn/supervisor/internal/triggers/utils/datasets/dataloader_info.py

```diff
@@ -1,30 +1,20 @@
 # TODO(415): Unify with similar classes in trainer_server and evaluator
+
+from dataclasses import dataclass
+
+
+@dataclass
 class DataLoaderInfo:
-    def __init__(
-        self,
-        pipeline_id: int,
-        dataset_id: str,
-        num_dataloaders: int,
-        batch_size: int,
-        bytes_parser: str,
-        transform_list: list[str],
-        storage_address: str,
-        selector_address: str,
-        num_prefetched_partitions: int,
-        parallel_prefetch_requests: int,
-        shuffle: bool,
-        tokenizer: str | None,
-    ):
-        self.pipeline_id = pipeline_id
-        self.dataset_id = dataset_id
-        self.num_dataloaders = num_dataloaders
-        self.batch_size = batch_size
-        self.bytes_parser = bytes_parser
-        self.transform_list = transform_list
-        self.storage_address = storage_address
-        self.selector_address = selector_address
-        self.num_prefetched_partitions = num_prefetched_partitions
-        self.parallel_prefetch_requests = parallel_prefetch_requests
-        self.tokenizer = tokenizer
-        self.training_id = -1
-        self.shuffle = shuffle
+    pipeline_id: int
+    dataset_id: str
+    num_dataloaders: int
+    batch_size: int
+    bytes_parser: str
+    transform_list: list[str]
+    storage_address: str
+    selector_address: str
+    num_prefetched_partitions: int
+    parallel_prefetch_requests: int
+    shuffle: bool
+    tokenizer: str | None = None
```

Comment:
there was no default for this before - are we sure nothing breaks by adding a default here?

Reply:
What's the point of allowing `None` while not using it as the default? If any part of the code didn't specify the argument before, this was already an undefined behavior and should have been flagged by mypy. If `None` breaks something here, we should also not allow it to be passed.

Reply:
fair